src/librustc_codegen_utils/llvm_target_features.rs

```diff
@@ -0,0 +1,112 @@
+use rustc::session::Session;
```

Comment:
Moved to `rustc_codegen_utils`, because `MetadataOnlyCodegenBackend` needs to pass them to `target_features_whitelist` to compile `core`.

Reply:
I don't think this is a good direction - LLVM shouldn't be treated that special. Is there a way to make `core` not contain a usage `core::arch` if the target features are missing?

Reply:
@eddyb what do you mean here with "if the target features are missing"? Because of run-time feature detection, `core::arch` always need to be available.

> I don't think this is a good direction - LLVM shouldn't be treated that special

Once we add cretonne as a backend we'll need to map Rust's feature names to backend names (once for LLVM, once for Cretonne), and all backends won't support all targets. So I think that at least the white-lists are going to need to exist on a per backend basis.